wp-content/themes/chrisdinh-photography/app/View/Composers/PageGallery.php

```diff
@@ -14,13 +14,24 @@ class PageGallery extends Composer {
     ];
 
     private function getGalleryItems() {
-        $posts = get_posts([
-            'post_type' => 'gallery_items',
-            'post_status' => 'publish',
-            'posts_per_page' => -1
-        ]);
+        $allItems = get_field('gallery__gallery-images');
 
-        return $posts;
+        $items = [];
+
+        foreach ($allItems as $item) {
+            $terms = get_the_terms($item['ID'], 'collection');
+
+            // If we have any collections set, add them to our new items array
+            if ( is_array($terms) && !empty($terms) ) {
```

Comment:
Consider modifying to handle cases where $terms is false or null.

false: When the taxonomy doesn't exist or an error occurs.
null: When the taxonomy exists, but no terms are assigned to the post.

suggestion:
// Check if $terms is valid, an array, and not empty
if ( $terms && is_array($terms) && !empty($terms) ) {

Reply:
`empty()` evaluates `null` and `false` already so we're doing that currently! [empty() php documentation](https://www.php.net/manual/en/function.empty.php). The one use case i have not handled here is if `get_the_terms` returns `WP_Error` - i've added a conditional for that in my [latest commit ](https://github.com/jonathanmdinh/itschrisdinh.com/pull/6/commits/a53d1d7a6ac8c8c33642df4ff1327a4b1ce2998e) @jonathanmdinh

Reply:
Very nice 👍